.github/workflows/build-gpu.yml

```diff
@@ -35,9 +35,7 @@ jobs:
       - name: Install dependencies
         run: |
           conda install -c conda-forge gsl
```

Comment:
Can delete gsl here also, it's not needed now that msprime 1.0 is out. (Has been removed from other workflows, AFAIK)

Reply:
Removed.

Test passes: https://github.com/aktech/sgkit/actions/runs/949300393